### PR TITLE
Convert code base to ROS2 Foxy

### LINKED ISSUE #109: support for ROS2
A meta-issue for supporting ROS2. ROS2 is slowly becoming the new default and folks will be trying to switch over over the next months/years. This is a good opportunity to become familiar with the new standard and attract more developers.

Feel free to chime in with thoughts.

### PR DESCRIPTION
**Pull request by Achllle**
close #109 

Note the target branch. At a later date we can make foxy-devel the default branch and do backports to `melodic-devel` and `noetic-devel`.

See commits and the ROS2 tutorials to make sense of the changes.

## TODO

- [x] Change installation instructions to Ubuntu20.04
- [x] Check systemd scripts
- [ ] ~~Led screen version~~
- [x] convert joy launch

### CODE REVIEW

**Comment by Achllle** [comment]
I'm probably not going to implement the LED screen as I don't have it.

**Comment by ericjunkins** [comment]
I can't currently test this without a rover, and am not particularly familiar with ROS2  yet, but I looked over the files and have a relative sense of this and looks good. If there's something specific we want to discuss happy to, but I think otherwise we move forward with adding it as its' own supported branch.

Nice job 👍

**Review by ericjunkins — APPROVED** [review]

**Comment by apollokit** [comment]
Some suggested changes...

but also, roboclawtest.py is failing for me (section 6 of https://github.com/Achllle/osr-rover-code/blob/iss109_ros2foxy/setup/rpi.md, after going through all the steps of setting up with 20.04)

I'm assuming this is because of the update to ubuntu 20.04. Are you able to run the script successfully?

**Comment by Achllle** [comment]
It should work. Are you running it with Python3? (The documentation should be updated to `python3` iso `python` but `python2` isn't installed on 20.04 by default. If you did install python2, remove it :)

**Comment by apollokit** [comment]
I did use python3....damnit, I feel like my setup has a ghost

**Review by apollokit — DISMISSED** [review]

**Comment by apollokit** [comment]
> I did use python3....damnit, I feel like my setup has a ghost

Ok, with my latest changes, talking to the roboclaw controllers successfully, and able to run ros2. Haven't yet gotten my xbox controller working with the rover, so want to look into that, but I think this PR is good.

File: ROS/osr_control/osr_control/rover.py [hunk 1]
```diff
@@ -48,25 +66,27 @@ def cmd_cb(self, twist_msg, intuitive=False):
         controlling the robot with a joystick or other manual input topic, consider using the 
         /cmd_vel_intuitive topic instead.
 
-        The Intuitive mode (intuitive=True) means that sending a positive angular velocity (moving joystick left)
+        The Intuitive mode (intuitive=True) means that sending a positive anguwlar velocity (moving joystick left)
```
Comment (Achllle) [inline ROS/osr_control/osr_control/rover.py:69]:
```suggestion
        The Intuitive mode (intuitive=True) means that sending a positive angular velocity (moving joystick left)
```

File: setup/rpi.md [hunk 2]
```diff
@@ -50,92 +52,77 @@ You probably will also want to connect to your newly configured RPi remotely ove
...
 ```
 
 ### 4.2 Clone and build the rover code
 For this section, you'll be working with the version control software `git`. Now's a good time to [read up](https://try.github.io/) on how that works if you're new to it and make a GitHub account!
 In the newly created catkin workspace you just made, clone this repo:
 ```commandline
-sudo apt-get install git
+sudo apt install git
 cd ~/osr_ws/src
 git clone https://github.com/nasa-jpl/osr-rover-code.git
```
Comment (Achllle) [inline setup/rpi.md:90]:
```suggestion
git clone https://github.com/nasa-jpl/osr-rover-code.git
cd osr-rover-code
git fetch origin
git checkout foxy-devel
```

**Pull request merged**